### DISCUSSION TITLE
Feature Request: Changing Settings

### DISCUSSION THREAD

**Discussion by ErickvanWyngaard** [StephanJoubert/home_assistant_solarman#90] [Ideas]
Hi,

Is it possible to add changing of settings from home assistant? Specifically the "Battery Low" setting (Register 219). I want to change when my battery is charged from the grid when I know the power will be going off.

Thanks

**Comment by torsteinelv** [comment]
Also interested in this. Think this can be activated on the deye inverter if you email deye and ask them to enable this. Then the modbus command can be sendt.

**Comment by torsteinelv** [comment]
Is this plugin sending modbus commands to the inverters? If so changing settings command would be possible to send as well?

**Comment by torsteinelv** [comment]
After version C35D deye have opened for remote control. Mabey this will be more easy to implement now?

**Comment by soren-norgaard** [comment]
A python module has been created which does that https://github.com/jmccrohan/pysolarmanv5 and then you could e.g. change the charge amp with this script. Would be awesome to have it inside this integration in home assistant. 


""" Scan Modbus registers to find valid registers"""
from pysolarmanv5 import PySolarmanV5
"""import requests
"""
def main():
    modbus = PySolarmanV5(
        "10.0.1.XX", 271XXXXXXX, port=8899, mb_slave_id=1, verbose=0
    )
    modbus.write_multiple_holding_registers(register_addr=108, values=[169])
    
if __name__ == "__main__":
    main()
